presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java

```diff
@@ -57,16 +57,28 @@ public StatisticAggregations(
         return groupingVariables;
     }
 
-    public Parts createPartialAggregations(PlanVariableAllocator variableAllocator, FunctionManager functionManager)
+    public Parts splitIntoPartialAndFinal(PlanVariableAllocator variableAllocator, FunctionManager functionManager)
```

Comment:
Due to my lack of knowledge about aggregation, I find changes in this class very confusing to me. Could you please let me know why we need to separate final and intermediate cases here? Can we actually have two extended class that share most code?